### PR TITLE
Addedd ESP8266 firmware code

### PR DESCRIPTION
**Pull request by iayanpahwa**
Signed-off-by: Ayan Pahwa (iayanpahwa@gmail.com)

Description: ESP8266 MQTT endpoint firmware.

DEPENDENCIES: 
1) Arduino MQTT client library (https://github.com/knolleary/pubsubclient)
2) ESP8266 WiFi library (https://github.com/ekstrand/ESP8266wifi)

LICENSE: MIT

TEST SUMMARY: Build Passed with Arduino IDE 1.6.11, Tested with Mosquito broker 1.3.4 running on Raspberry Pi 3 

KNOWN LIMITATIONS: None

### CODE REVIEW

**Comment by iayanpahwa** [comment]
@anshulkatta @embtux can you please review it.

**Comment by embtux** [comment]
@iayanpahwa ,

1. For GPLv3, use one of the statements from - https://www.gnu.org/licenses/gpl-howto.en.html.
For example, below message for one file program:
```
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

**Comment by teamsdiot** [comment]
@embtux 
1. I'll commit the License text in files.
2. The MIT License mention is for the dependencies source libraries, they are using MIT license, if I am supposed not to mention that we can remove LICENSE tag from PR summary

**Comment by teamsdiot** [comment]
@embtux changes have been commit.

File: firmware_ESP8266/esp_node.ino [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  // Loop until we're reconnected
+  while (!client.connected()) {
+
+    #if SERIAL_DEBUG
+    Serial.print("Attempting MQTT connection...");
+    #endif
+
+    // Attempt to connect
+    if (client.connect("ESP8266Client")) {
```
Comment (anshulkatta) [inline firmware_ESP8266/esp_node.ino]: If we have more than one Client then , the name would be same ?

Comment (iayanpahwa) [inline firmware_ESP8266/esp_node.ino]: Yes, that should not be an issue!

**Comment by embtux** [comment]
Visual Ack.

    Acked-by: Ashwini Singh <asingh.engi@gmail.com>

**Comment by embtux** [comment]
Don't have merge rights.
So, as of now, someone will have to take care of that.

**Comment by iayanpahwa** [comment]
@embtux I shall merge it with your ACK.

**Pull request merged**